netmaster/master/netmaster_test.go

```diff
@@ -296,3 +309,173 @@ func TestVxlanConfigWithLateHostBindings(t *testing.T) {
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net5",
+            "PktTag"            : 1200,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, true)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant6",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net6",
+            "PktTag"            : 1201,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, false)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant7",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net7",
+            "PktTag"            : 1500,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, false)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant8",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net8",
+            "PktTag"            : 1501,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, true)
+
+}
+
+func applyVerifyRangeTag(t *testing.T, cfgBytes []byte, shouldFail bool) {
+	initFakeStateDriver(t)
+	defer deinitFakeStateDriver()
+
+	cfg := &intent.Config{}
+	err := json.Unmarshal(cfgBytes, cfg)
+	if err != nil {
+		t.Fatalf("error '%s' parsing config '%s'\n", err, cfgBytes)
+	}
+
+	_, err = resources.NewStateResourceManager(fakeDriver)
+	if err != nil {
+		log.Fatalf("state store initialization failed. Error: %s", err)
+	}
+	defer func() { resources.ReleaseStateResourceManager() }()
+
+	tenant := cfg.Tenants[0]
+	err = CreateTenant(fakeDriver, &tenant)
+	if err != nil {
+		t.Fatalf("error '%s' creating tenant\n", err)
+	}
+
+	err = CreateNetworks(fakeDriver, &tenant)
+	if shouldFail {
+
+		var expError string
+		network := tenant.Networks[0]
+		if network.PktTagType == "vlan" {
+			expError = fmt.Sprintf("vlan %d does not adhere to tenant's vlan range %s", network.PktTag, tenant.VLANs)
+		} else {
+			expError = fmt.Sprintf("vxlan %d does not adhere to tenant's vxlan range %s", network.PktTag, tenant.VXLANs)
+		}
+
+		if err.Error() != expError {
```

Comment:
this might panic if err is nil, so may be we should check for `err == nil` as well?